tests/averaging/rotation/test_cycle_consistency.py

```diff
@@ -0,0 +1,304 @@
+"""Unit tests to ensure correctness of cycle triplet extraction and cycle error computation.
+
+Author: John Lambert
+"""
+import time
+from typing import Dict, List, Tuple
+
+import numpy as np
+from gtsam import Rot3, Unit3
+
+import gtsfm.averaging.rotation.cycle_consistency as cycle_utils
+from gtsfm.two_view_estimator import TwoViewEstimationReport
+
+
+def test_extract_triplets_adjacency_list_intersection1() -> None:
+    """Ensure triplets are recovered accurately.
+
+    Consider the following undirected graph with 1 cycle:
+
+    0 ---- 1
+          /|
+         / |
+        /  |
+      2 -- 3
+           |
+           |
+           4
+    """
```

Comment:
Can we add 2 more cycles: one which shares an edge with the current cycle and the other which just shares a node.

Reply:
So I think the other unit test includes the case where we share an edge with the current cycle.

I can add another that just shares a node.

Reply:
I added another unit test to check the shared node case.